### ISSUE TITLE
Offer ergonomics

### ISSUE THREAD

**Issue by PKEv** [QTodoTxt/QTodoTxt#282]
The difficulty arises with the use of tags when writing tasks, due to the lack of the @ symbol in my keyboard.
Can add more buttons to the dialog box "Add task" by which you can dial special characters?

**Comment by topas-rec** [comment]
Just found this one here and I think many programs cannot be used without a full keyboard layout.  (Some special keys would be missing somewhere - no email without the @ symbol)
Is really every program responsible for supplying soft-keys that it needs? Why can't one download a software keyboard with which one could handle every missing key in all kind of software?
If this is true I would want to close this issue.

**Comment by topas-rec** [comment]
I would also love a complete abstraction from todo.txt text format through the input dialog when creating a task. A part of this dialog could be the calendar picker in the way it is mentioned in #130.
Another part should be a tick list of already existing contexts (or a line for new ones). This dialog would also eliminate the need of a soft-@-button.

**Comment by PKEv** [comment]
What about this?
![default](https://cloud.githubusercontent.com/assets/6053197/20143836/976038f4-a6ab-11e6-8e73-47b89a43473e.PNG)

**Comment by oroulet** [comment]
My plan was to remove that dialog, not make it even bigger ;-).
You have done quite a lot of with lately, and I think that people doing the
work decide, but that change make me uncomfortable, there are very very
very few people needing such buttons...

**Comment by PKEv** [comment]
What do you mean by disposal of this dialogue, I do not understand?

**Comment by oroulet** [comment]
My plan was to remove that dialog, not make it even bigger ;-).
You have done quite a lot of work lately, and I think that people doing the
work decide, but that change make me uncomfortable, there are very very
very few people needing such buttons...
…

**Comment by oroulet** [comment]
1) My goal is to remove this dialog. The edit/add can be done directly in
listview.
2) 99%of people do not need these buttons

**Comment by PKEv** [comment]
In such a case, it may make the extended context menu for the input task field?

**Comment by topas-rec** [comment]
@oroulet Can you explain your point no 1 in your last comment?
How should the integration in the list view work?

**Comment by topas-rec** [comment]
I love the way simple tasks abstracts from the text entry by providing small dialogs:

![tmp_5127-screenshot_20161110-094807-467235396](https://cloud.githubusercontent.com/assets/20391449/20170023/e99f3e6a-a72a-11e6-83af-5d957164bfb2.png)
![tmp_5127-screenshot_20161110-094828-580947680](https://cloud.githubusercontent.com/assets/20391449/20170036/f493453c-a72a-11e6-85b1-ac35ad9083f0.png)
![tmp_5127-screenshot_20161110-094814238024429](https://cloud.githubusercontent.com/assets/20391449/20170039/f72af448-a72a-11e6-8951-2a0219b0e816.png)

**Comment by oroulet** [comment]
Such inline dialogs are no issue. We already have a dialog for completion.
But I would like not to use a floating dialog for editing, but use the
editor integrated in list view. From the ui design point of view it give a
cleaner interface with editing in context of other tasks

**Comment by topas-rec** [comment]
Do you mean something like inline editing?
I image working with your suggestion in the following way:
I just have a list of tasks on the right pane. I can change the priority of a single task if I click on the priority of that task. Then a drop-down menu opens and I can select the new priority.
In the same way there could be a drop-down for contexts or projects (allowing multiple selections)
Is that the way you think of it? Sorry if I didn't get it yet.

**Comment by topas-rec** [comment]
If so, sorting and detailed filtering also is possible by design because the single column list becomes a multiple column table.
(Assuming that in a table UI of Qt these features are present)

**Comment by PKEv** [comment]
What about the pop-up menu with the ability to:
installation of special characters: +, @, due:, t:
Formatting: <br>, <u>, <b>
???

**Comment by oroulet** [comment]
If you really want it.... But make it an option....

**Comment by topas-rec** [comment]
Do you mind to have drop down menus for priority @PKEv instead of buttons? They can be filled with already used values.
I also could imagine to have them for:
- Context
- Project
And a calendar selector (http://doc.qt.io/qt-5/qml-qtquick-controls-calendar.html) would be nice to have for due and threshold.

As mentioned above I think this should work on the task list itself too (if we want to get rid of the pop-up)

**Comment by oroulet** [comment]
But we already have completion, don't we? It is a kind of dynamic drop down
menu

**Comment by topas-rec** [comment]
Of course we have. I agree with you @oroulet. But to use it one has to know the special characters of the todo.txt format.
Inexperienced users might not be aware of these functions (contexts, due dates and ...).
If there were buttons or drop down menus the user would know there is something more than just an empty text line for a task name.
By having buttons the user is more guided through the program. That's what (i think) a GUI is made for. In my opinion a good GUI can make a handbook or help manual completely unnecessary.

I don't want to dictate anything. Just want to join the discussion.

**Comment by oroulet** [comment]
I agree that discoverability is an argument for having such buttons....

**Comment by oroulet** [comment]
> I just have a list of tasks on the right pane. I can change the priority of a single task if I click on the priority of that task.

With inline editing, if you click on a task you get directly into edit mode and if one click on a key, you will get the floating popup we allready have to display possible alternatives. No extra colums.

I am really not sure where to place button to increase discoverability of syntax with + @ , due:, etc maybe create dynamically small buttons on the right side of editor field

**Comment by topas-rec** [comment]
>With inline editing, if you click on a task you get directly into edit mode 
I see. And I like it.

Why not implementing the inline mode and an pop-up editor as an option? (For inexperienced users the dialog should be enabled by default)

**Comment by oroulet** [comment]
I tried but did not make it. We need first to fix the editdialog implementation. Make the dialog a normal dialog and not a global one that hides and shows. Maybe you are interested to have a look at this @PKEv ?

**Comment by oroulet** [comment]
Now I implemented the inline editor (as an option), so if someone really want to add buttons to the big dialogs, they  can. I do not care ;-)
